packages/button/index.js

```diff
@@ -63,64 +63,77 @@ class WarpButton extends kebabCaseAttributes(WarpElement) {
     }
   }
 
-  get _classes() {
+  get _primaryClasses() {
     const primary = this.variant === 'primary';
+    return [
+      primary && !this.small && !this.quiet && !this.loading && ccButton.primary,
+      primary && this.small && !this.quiet && !this.loading && ccButton.primarySmall,
+      primary && this.small && this.quiet && !this.loading && ccButton.primarySmallQuiet,
+      primary && this.small && this.loading && (this.quiet ? ccButton.primarySmallQuietLoading : ccButton.primarySmallLoading),
+      primary && !this.small && this.quiet && !this.loading && ccButton.primaryQuiet,
+      primary && !this.small && this.loading && (this.quiet ? ccButton.primaryQuietLoading : ccButton.primaryLoading),
+    ];
+  }
```

Comment:
Just an idea I thought of now. Maybe we could move the `primary` condition to where `this._primaryClasses` is referenced, and so make the above conditions shorter + not repeat ourselves within the context of `_primaryClasses` getter? So something like this:
```
get _primaryClasses() {
  return [
    !this.small && !this.quiet && !this.loading && ccButton.primary,
    this.small && !this.quiet && !this.loading && ccButton.primarySmall,
    ...
  ]}
  
  
get _classes() {
  return classNames(this.buttonClass, [
      this.variant === 'primary' && ...this._primaryClasses,
      this.variant === 'secondary' && ...this._secondaryClasses,
      ...
  ])}
  ```

Reply:
Ah nice! I'll change it to that :)

Reply:
I have to change it like this instead 
`get _classes() {
    return classNames(this.buttonClass, [
      ...(this.variant === 'primary' ? this._primaryClasses : []),
      ...(this.variant === 'secondary' ? this._secondaryClasses : []),
      ...(this.variant === 'utility' ? this._utilityClasses : []),
      ...(this.variant === 'negative' ? this._negativeClasses : []),
      ...(this.variant === 'pill' ? this._pillClasses : []),
      ...(this.variant === 'link' ? this._linkClasses : []),
      this.fullWidth ? ccButton.fullWidth : ccButton.contentWidth,
    ]);
  }`

Otherwise it throws an error: "Expression expected"

Reply:
Updated in this commit: [26ae621](https://github.com/warp-ds/elements/pull/174/commits/26ae621926d669d1ebb9edc98a2cae74b10c09c6)

Reply:
We can actually do `this.variant === 'primary' && this._primaryClasses`. When I logged `this._classes` in the console, it showed a correct string. Seems like `classNames` function handles nested arrays.

Reply:
Updated in this commit: [d6e1bf7](https://github.com/warp-ds/elements/pull/174/commits/d6e1bf724f3b23ad0801112f545a9327332ab033)